Create new user-friendly view

lmcmicu opened the pull request:
Potentially closes #58 (if we decide on this solution). Note that this branch has been forked from the [undo-redo branch](https://github.com/ontodev/valve.rs/tree/undo-redo) (which is the reason why the diff shows changes other than the one in https://github.com/ontodev/valve.rs/pull/60/commits/b4484ff477628e0cb9082517fca6f01682b0be6e (which adds the new view)).

Before merging this PR, PR #59 must be merged to main first and then this branch should be rebased onto main. I am leaving this PR in draft mode to prevent inadvertently merging it before this has been done.

lmcmicu commented:
This is ready for review as far as I'm concerned, but I am leaving the PR in draft mode since #59 must be merged first (see the description of this PR).

jamesaoverton commented:
This has been working for me so far. I would like two changes, please:

1. Instead of `*_user_view` I would like to call this `*_text_view`.
2. I'm finding the names `get_last_undo()` and `get_last_change()` confusing. I'd prefer something like `get_undo()` and `get_redo()`.

lmcmicu commented:
Done. I've changed `user_view` to `text_view`. I've also changed `get_last_undo()` to `get_record_to_redo()` and `get_last_change()` to `get_record_to_undo()`
